File: BookLibraryApi/Models/Book.cs
using System;

namespace BookLibraryApi.Models;

public class Book
{
    public string Title { get; set; }
    public string Author { get; set; }
    public string ISBN { get; set; }
    public int Year { get; set; }

    Book(string _ISBN = "", string _title = "", string _author = "", int _year)
    {
        ISBN = _ISBN;
        Title = _title;
        Author = _author;
        Year = _year;
    }
}

File: workshop-react-library/src/types/book.ts
export interface Book {
  title: string;
  author: string;
  isbn: string;
  year: number;
}

export interface CreateBookDto {
  title: string;
  author: string;
  isbn: string;
  year: number;
}

export interface UpdateBookDto {
  title?: string;
  author?: string;
  year?: number;
}

// export interface BookQueryParameters {
//   author?: string;
//   year?: number;
// }

// export interface PaginatedResponse<T> {
//   data: T[];
//   pagination: {
//     page: number;
//     pageSize: number;
//     totalCount: number;
//     totalPages: number;
//   };
// }
